core/src/main/java/com/github/jsonldjava/utils/JsonUtils.java

```diff
@@ -371,6 +363,56 @@ public static Object fromURL(java.net.URL url, CloseableHttpClient httpClient)
         }
     }
 
+    private static InputStream getJsonLdViaHttpUri(final URL url, final CloseableHttpClient httpClient,
+            CloseableHttpResponse response) throws IOException {
+        final HttpUriRequest request = new HttpGet(url.toExternalForm());
+        // We prefer application/ld+json, but fallback to application/json
+        // or whatever is available
+        request.addHeader("Accept", ACCEPT_HEADER);
+        response = httpClient.execute(request);
```

Comment:
This will not work in Java to close the response in the outer method. You will need to do the try-catch or try-with-resources in here.

Reply:
Treated with 776e77dc63612ef294b814e3ad5d25fddf6e8e1e and 818b118d23180385ad9ab667c8e1a022984e9fa5 .

Reply:
Comment about another possible resource leak in the latest version of this.